src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -143,6 +239,17 @@ void AggregatedDataVariants::convertToTwoLevel()
     }
 }
 
+void AggregatedDataVariants::serialize(WriteBuffer & wb, const Aggregator & aggregator_) const
+{
+    aggregator_.checkpoint(*this, wb);
+}
+
+void AggregatedDataVariants::deserialize(ReadBuffer & rb, const Aggregator & aggregator_)
+{
+    aggregator = &aggregator_;
```

Comment:
I feel like we shall separate this assignment and to make it explicitly and get called else where before `deserialize(...)`. Basically it is not good to assign aggregator here